dashboard/app/controllers/pd/workshop_daily_survey_controller.rb

```diff
@@ -383,17 +384,30 @@ def new_csf_post201_foorm
     def thanks
     end
 
+    # Pre survey controller for academic year workshops
+    def new_ayw_pre
+      ayw_helper(PRE_SURVEY_CONFIG_PATHS, should_have_attended: false, day: 0)
+    end
+
+    def new_ayw_daily
+      ayw_helper(DAILY_SURVEY_CONFIG_PATHS, should_have_attended: false, day: params[:day])
+    end
+
+    def new_ayw_post
+      ayw_helper(POST_SURVEY_CONFIG_PATHS, should_have_attended: true, day: nil)
+    end
+
     protected
 
-    def get_attended_workshop_by_enrollment_or_course(enrollment_code, course, subject)
-      attended_workshop = nil
+    def get_workshop_by_enrollment_or_course(enrollment_code, course, subject, should_have_attended=true)
```

Comment:
This is pretty logically tricky and I don't se any unit tests, might be good to add as a follow up?

Reply:
good call, I can add it now since it's used multiple places

Reply:
Writing it out to wrap my head around it...

If an enrollment code is provided, we search for a workshop by that enrollment code. If that enrollment code is invalid, we render invalid enrollment code. 

If the current user attended the workshop we found, or we don't care whether the user attended the workshop (I don't understand this part), we set this workshop as the workshop to return.

If no enrollment code is provided, we get all workshops that this user has enrolled in. If they haven't enrolled in any workshops, we render not enrolled.

If they have enrolled in workshops AND we care that they attended the workshop, we set the workshop to return to the most recent attended workshop; otherwise, we return the most recent workshop they're enrolled in.

If we have no workshops at this point, then we couldn't find a workshop with attendance for this user.

Reply:
I'll add a comment explaining this :) The should_have_attended addition covers that for pre surveys we don't expect the user to be marked attended yet and it is not a prerequisite for them filling out the survey. For post surveys we do want the user to have been marked attended.
Scenarios this covers:
- user provided an enrollment code-we find the enrollment with that code. If they need to have attended, we verify the user attended the workshop.
- user did not provide an enrollment code: we find the most workshops the user has enrolled in. If we don't care about attendance, we find the most recent workshop. If we do care about attendance, we find the most recent workshop that they were marked as attended. 

If at the end the workshop_to_return was blank we know it's because there was no attendance, because we already checked that there were possible workshops they were enrolled in.

Reply:
turns out its hard to add a test for a helper method in a controller in isolation--I'll update the comment here and make sure the unit tests for this method exercise the functionality

Reply:
I think I've found that in the past, too, and have just hoisted functions into the model where they are much easier to test.  (And therefore probably belong there :)

Reply:
👍 I will refactor this as a follow-up (there's a lot in here that I think could be refactored to the model but it's out of scope for this pr)